src/components/3d_editor/advanced_geometry/python_transformation/JupyterLiteTransformation.tsx

```diff
@@ -50,33 +55,38 @@ class JupyterLiteTransformation extends React.Component<
             // eslint-disable-next-line react/no-did-update-set-state
             this.setState({ materials });
         }
+        this.messageHandler.sendData(this.returnSelectedMaterials());
     }
 
-    componentWillUnmount() {
-        window.removeEventListener("message", this.handleReceiveMessage, false);
-    }
+    returnSelectedMaterials = () => {
+        const { selectedMaterials } = this.state;
+        return selectedMaterials.map((material) => material.toJSON());
+    };
 
-    handleReceiveMessage = (event: MessageEvent) => {
-        // Check if the message is from the expected source
-        // TODO: check for partial URL match, e.g. with "/" at the end
-        if (event.origin !== ORIGIN_URL) {
-            return;
-        }
-        if (event.data.type === "from-iframe-to-host") {
-            try {
-                // TODO: add check for Material config type
-                const configs = event.data.data.materials;
-                if (Array.isArray(configs)) {
-                    this.setState({
-                        newMaterials: configs.map((config) => new Made.Material(config)),
-                    });
+    setMaterials = (data: any) => {
```

Comment:
set materials should just set materials with setState, this is much more complicated function. Should be called handleSetData or something like that

Reply:
Ok